test/distributed/org/apache/cassandra/distributed/upgrade/ClusterMetadataUpgradeHarryTest.java

```diff
@@ -56,7 +59,28 @@ public class ClusterMetadataUpgradeHarryTest extends UpgradeTestBase
     public void simpleUpgradeTest() throws Throwable
     {
         AtomicReference<Interruptible> executor = new AtomicReference<>();
+        AtomicLong loops = new AtomicLong(0);
         Listener listener = new Listener();
+        Runnable awaitHarryProgress = () -> {
+            long startingLoopCount = loops.get();
+            long deadline = startingLoopCount + 100;
```

Comment:
100 was randomly selected... on my m1 max laptop 100 loops is ~1s... so this is less progress than previous.... our definition of work is also different as we track the read/write so we know 100 writes have at least happened... which feels fine for this test?